src/app/_guards/auth.guard.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, ActivatedRouteSnapshot } from '@angular/router';
+import { AuthService } from '../services/auth.service';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+    constructor(
+        private authService: AuthService,
+        private router: Router) {}
+
+    canActivate(next: ActivatedRouteSnapshot): boolean {
+        const roles = next.data['roles'] as Array<string>;
+        if (roles) {
+            const match = this.authService.roleMatch(roles);
+            if (match) {
+                return true;
+            } else {
+                this.router.navigate(['last-7-day-post-list']);
+                window.alert('You are not authorized to access this area!');
+            }
+        }
+
+        if (this.authService.loggedIn()) {
+            return true;
+        }
+
+        window.alert('You are not logged in!');
+        this.router.navigate(['last-7-day-post-list']);
+        return false;
+    }
+}
```

Comment:
Alerts are a bit aggressive. We don't have any error message handling in place yet but I think leaving the alerts off for now is ok (maybe have a console log) and once we have error message handling we can then inform users why they're being kicked back to the last 7 days list.

Reply:
Cool will do.